File: pet-eventos-frontend/src/app/model/Event.ts
import { User } from './User';
import { Attendance } from './Attendance';
import { Section } from './Section';
export class Event {
  idEvent: number;
  name: String;
  sections: Section[];
  users: User[]
  description: String;
  place: String;
  isActive: boolean;
  active: boolean;
  cancelled: boolean;
  isCancelled: boolean;
  attendances: Attendance[];
}

File: pet-eventos-backend/src/main/java/com/rafael/peteventos/domain/entity/Event.java
package com.rafael.peteventos.domain.entity;

import lombok.*;

import javax.persistence.*;
import java.util.List;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder(setterPrefix = "with")
public class Event {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idEvent;

    @Column
    private String name;

    @Column
    private String description;

    @Column
    private boolean isActive;

    @Column
    private boolean isCancelled;

    @Column
    private String place;

    @ManyToMany
    private List<User> users;

    @OneToMany(mappedBy = "event", cascade = CascadeType.ALL)
    private List<Certificate> certificates;

    @OneToMany(mappedBy = "event", cascade = CascadeType.ALL)
    private List<Section> sections;

    @ManyToOne(fetch = FetchType.LAZY)
    private Group group;

    private Long total;

}
